### ISSUE TITLE
support database import from other password managers, csv

### ISSUE THREAD

**Issue by nickolasclarke** [mozilla-lockbox/lockbox-addon#133]
I would love to migrate from dashlane to lockbox as my primary password management too. Support for importing passwords from 3rd party tools such as 1pass, dashlane and a simple csv format would be helpful to drive adoption.

**Comment by maverick74** [comment]
Beside **import** from csv, I would like an **export** to csv option as well

Should i open a new bug for this?

**Comment by SilentFez** [comment]
Until it is implemented you can use this tool https://github.com/louisabraham/ffpass
I just tried it with a csv from bitwarden and it imported my vault flawlessly.

**Comment by Alhaitham** [comment]
Import an export feature would be great.

**Comment by Alhaitham** [comment]
Import and export feature would be great.

**Comment by da2x** [comment]
Import and export tools are required by the _GDPR Article 20 Right to data portability_.

**Comment by A-Charvin** [comment]
Was about to report the same thing, An Import/Export function is much needed to get more users using other password managers to switch over or to consider or to even properly test out lockwise.

**Comment by Javierkaiser** [comment]
Yes, it would be nice, I was about to change to Lockwise, but wasn't able to import the passwords from my actual password manager.

**Comment by andreicristianpetcu** [comment]
Both import and export landed. dashlane
https://github.com/mozilla-lockwise/lockwise-addon/issues/332#issuecomment-653938961
![image](https://user-images.githubusercontent.com/922368/86542248-ad8e6000-bf1c-11ea-96c6-5d0d0da17b66.png)

I did not test Dashlane passwords CSV file. What files does it have?

**Comment by andreicristianpetcu** [comment]
On stable you need to open devtools, search for `menuitem-import` and then remove the `hidden=""`
![image](https://user-images.githubusercontent.com/922368/96098276-988d7980-0eda-11eb-8d7b-fe444fe84044.png)

The reason for this is that UX really wanted a fancy import dialog and I did not have time to implement it. I believe in incrementalism, UX believes in perfection.

**Comment by maverick74** [comment]
@andreicristianpetcu 
So, instead of giving us a useful feature that a lot of people needs, UX decided to not allowing us to use it because it wants a fancy dialog....... I'm sure that was **_!_** the right move (notice the ! = not?! LOL)

I went into **about:config** and set **signon.management.page.fileImport.enabled** to **true**

**Comment by andreicristianpetcu** [comment]
My initial implementation had no feedback to the user. A failed import was not reported in any way. It's not just UI. It's quite important to know if a password was imported or if it had an error while a row was processed.

I think I was not clear in my previous comment that it's not just pretty UX, it's also functionality. Sorry about that.

**Comment by maverick74** [comment]
@andreicristianpetcu Oh! In that case i must agree that UI is important. Indeed i got the idea it was just missing the "pretty lights".

That kind of alerts are definitely very important.

Is there any ETA for this?

**Comment by andreicristianpetcu** [comment]
No ETA, sorry. I really want to ship this.

**Comment by andreicristianpetcu** [comment]
Import form CSV has landed in Nightly and unless we find some critical bug, we intend to ship it in v86 :crossed_fingers: 
Please test.

**Comment by andreicristianpetcu** [comment]
Password import is on by default on all channells. It will be available with v88 in March on Beta and in April for stable.
